knapsack/knapsack.py

```diff
@@ -1,43 +1,46 @@
 """ A naive recursive implementation of 0-1 Knapsack Problem
     https://en.wikipedia.org/wiki/Knapsack_problem
 """
-from __future__ import annotations
 
 
-def knapsack(capacity: int, weights: list[int], values: list[int], counter: int) -> int:
+def knapsack(capacity, weights, values, permit_repetition=False):
```

Comment:
I have only read this far and I am already convinced that CONTRIBUTING.md was not read.  Closing.